apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts

```diff
@@ -36,59 +36,11 @@ const validateDates = (event: IWebinarEvent): boolean => {
   return true;
 };
 
-const getLocale = async (event: IWebinarEvent): Promise<string> => {
-  // eslint-disable-next-line functional/no-let
-  let currentLocale: string | undefined;
-  if (event.params.data.locale) {
-    currentLocale = event.params.data.locale;
-  } else if (event.params.where?.id) {
-    const webinar: IWebinar | undefined = await strapi
-      .service('api::webinar.webinar')
-      .findOne(event.params.where?.id);
-    currentLocale = webinar?.locale;
-  }
-  return currentLocale || 'it';
-};
-
-const validateSlugUniqByLocale = async (
-  event: IWebinarEvent
-): Promise<boolean> => {
-  const currentLocale = await getLocale(event);
-
-  const query: { where: Record<string, unknown> } = {
-    where: {
-      locale: currentLocale,
-      slug: event.params.data.slug,
-    },
-  };
-  if (event.params.where?.id) {
-    query.where = {
-      id: {
-        $notIn: [event.params.where.id],
-      },
-      ...query.where,
-    };
-  }
-
-  const webinarWithSameSlug: IWebinar | undefined = await strapi.db
-    .query('api::webinar.webinar')
-    .findOne(query);
-
-  if (webinarWithSameSlug) {
-    throw new errors.ApplicationError(
-      'Webinar with the same slug already exists for the current locale'
-    );
-  }
-  return true;
-};
-
 module.exports = {
   async beforeCreate(event: IWebinarEvent) {
```

Comment:
Is `async` still necessary?
```suggestion
  beforeCreate(event: IWebinarEvent) {
```